matchtree.go

```diff
@@ -866,17 +870,32 @@ func (d *indexData) newMatchTree(q query.Q) (matchTree, error) {
 			return &noMatchTree{"const"}, nil
 		}
 	case *query.Language:
-		code, ok := d.metaData.LanguageMap[s.Language]
-		if !ok {
-			return &noMatchTree{"lang"}, nil
+		if d.metaData.IndexFeatureVersion < 12 {
+			// This is an old index file without more precise language mappings,
+			// assume that the query has file include patterns to approximate the
+			// desired language and match everything for lang: while waiting for
+			// the reindex to complete.
+			//
+			// TODO(rmmh): remove this code when reindex is complete and there
+			// are no more index files where this condition is true.
+			return &bruteForceMatchTree{}, nil
```

Comment:
I think we should use go-enry to generate the corresponding file match stuff? This way lang continues to work in zoekt without the assumption that sourcegraph is passing along those filters. Sourcegraph is no longer the only user of this repository.

Additionally we somehow need to correspond with your logic in `indexData.simplify` for ambigious files. Or we just don't support ambigious files. I prefer the latter.